src/client.ts

```diff
@@ -1604,6 +1604,9 @@ export class StreamChat<
    * @returns {Promise<APIResponse>}
    */
   async banUser(targetUserID: string, options?: BanUserOptions<UserType>) {
+    if (options?.user_id !== undefined) {
+      options.moderator_id = options.user_id;
```

Comment:
should we delete the `user_id` from the options if we replace it with `moderator_id`?

Reply:
I marked field as deprecated to avoid backwards incompatible changes. I'm not very experienced in TypeScript, so I'm not sure if field removal will break SDK users' code. If it's not, then probably would be sane to do that

Reply:
I mean deleting the `user_id` from the request  body before sending it to API, so you wouldn't have a request with `{user_id, moderator_id}`, however, this will make the backward compatibility test partially obsolete

Reply:
Ah, yes, we probably should. Would be the best to narrow things as much as possible to be sure that everything works with new approach.